4.applications/grit-informed-aggregation/1.generate-profiles/scripts/nbconverted/2.calculate-replicate-reproducibility.py

```diff
@@ -0,0 +1,258 @@
+        gg.aes(
+            fill="Metadata_agg_method", x="Metadata_agg_method", y="similarity_metric"
+        ),
+    )
+    + gg.geom_boxplot()
+    + gg.facet_wrap("Metadata_cell_line")
+    + gg.labs(title="Guide Replicate Correlation in Profiles")
+    + gg.labs(
+        x="Aggregation Method",
+        y="Replicate Pearson Correlation",
+        fill="Aggregation Method",
+    )
+)
+display(box)
+
+jitter = (
+    gg.ggplot(
+        guide_rep_cors.query("~Metadata_agg_method.str.contains('ctrls')"),
+        gg.aes(
+            fill="Metadata_agg_method", x="Metadata_agg_method", y="similarity_metric"
+        ),
+    )
+    + gg.geom_jitter()
+    + gg.facet_wrap("Metadata_cell_line")
+    + gg.labs(title="Guide Replicate Correlation in Profiles")
+    + gg.labs(
+        x="Aggregation Method",
+        y="Replicate Pearson Correlation",
+        fill="Aggregation Method",
+    )
+)
+display(jitter)
+
+
+# In[6]:
+
+
+box = (
+    gg.ggplot(
+        gene_rep_cors.query("~Metadata_agg_method.str.contains('ctrls')"),
+        gg.aes(
+            fill="Metadata_agg_method", x="Metadata_agg_method", y="similarity_metric"
+        ),
+    )
+    + gg.geom_boxplot()
+    + gg.facet_wrap("Metadata_cell_line")
+    + gg.labs(title="Gene Replicate Correlation in Profiles")
+    + gg.labs(
+        x="Aggregation Method",
+        y="Replicate Pearson Correlation",
+        fill="Aggregation Method",
+    )
+)
+display(box)
+
+jitter = (
+    gg.ggplot(
+        gene_rep_cors.query("~Metadata_agg_method.str.contains('ctrls')"),
+        gg.aes(
+            fill="Metadata_agg_method", x="Metadata_agg_method", y="similarity_metric"
+        ),
+    )
+    + gg.geom_jitter()
+    + gg.facet_wrap("Metadata_cell_line")
+    + gg.labs(title="Gene Replicate Correlation in Profiles")
+    + gg.labs(
+        x="Aggregation Method",
+        y="Replicate Pearson Correlation",
+        fill="Aggregation Method",
+    )
+)
+display(jitter)
+
+
+# #### Save outputs
+
+# In[7]:
+
+
+guide_rep_cors.to_csv(
+    Path(results_folder + "guide_replicate_correlations.tsv"), index=False, sep="\t"
+)
+gene_rep_cors.to_csv(
+    Path(results_folder + "gene_replicate_correlations.tsv"), index=False, sep="\t"
+)
+
+
+# ## Calculate percent replicating and percent matching
+
+# In[12]:
+
+
+random.seed(9000)
+
+n_samples = 1000  # Number of points to sample from the null distribution.
+n_replicates = 2  # Number of replicates of each sample.
+
+replicate_grouping_feature = "Metadata_pert_name"
+class_grouping_feature = "Metadata_gene_name"
+
+
+# ### Percent Replicating (previously percent strong)
+#
+# A "strong signature" is determined by :
+# 1. high pairwise (pearson) correlation w/ each other
+# 2. median replicate correlation is "significantly different" (aka perturbation with median replicate correlation greater than 95th percentile) from null distribution (pairwise correlations of non-replicates)
+
+# In[13]:
+
+
+get_ipython().run_cell_magic(
```

Comment:
What is the rationale for using:

```python
n_samples = 1000 # Number of points to sample from the null distribution.
n_replicates = 2 # Number of replicates of each sample.
```

The `n_replicates` is particularly interesting to me. Why only 2? if we're taking the median, isn't that effectively 1 then, except worse since you're taking the high value each time? Maybe this is why percent matching is so much worse!!!

Reply:
oh, did we figure this out? it's hard for me to see @hillsbury

Reply:
`n_replicates = 2` is the original number that Niranj used since I assume there are only two replicate for each compound that is tested. In our case, most guides only had 2 replicates per plate so I kept it the same but it should probably be 6 (since we want percent replicating across the experiment?). I tried both I think locally (so these are not the exact plots) and this number does impact the results. @gwaygenomics  Should I update before merge?

Reply:
yeah, i think we want it to be 6 - we are calculating here based on all perturbations for a given cell line, right?

Reply:
if you agree, please update before I merge. Thanks!

Reply:
how are we doing on this? I'd love to get this merged in!

Reply:
yes agreed and updated now--merge away!

Reply:
ok! gonna merge now